diff --git a/themes/claro/templates/form/field-multiple-value-form.html.twig b/themes/claro/templates/form/field-multiple-value-form.html.twig
index 58bfe31c..60610275 100644
--- a/themes/claro/templates/form/field-multiple-value-form.html.twig
+++ b/themes/claro/templates/form/field-multiple-value-form.html.twig
@@ -42,7 +42,7 @@
       <div{{ description.attributes.addClass(description_classes) }} >{{ description.content }}</div>
     {% endif %}
     {% if button %}
-      <div class="form-actions">{{ button }}</div>
+      <div class="field-actions">{{ button }}</div>
     {% endif %}
   </div>
 {% else %}
