diff --git a/plugin.js b/plugin.js
index 2eaecd4..826f28f 100644
--- a/plugin.js
+++ b/plugin.js
@@ -97,6 +97,9 @@
 			CKEDITOR.dialog.add( 'anchor', this.path + 'dialogs/anchor.js' );
 
 			editor.on( 'doubleclick', function( evt ) {
+        if (typeof editor.commands.drupallink !== 'undefined') {
+          return;
+        }
 				// If the link has descendants and the last part of it is also a part of a word partially
 				// unlinked, clicked element may be a descendant of the link, not the link itself (https://dev.ckeditor.com/ticket/11956).
 				// The evt.data.element.getAscendant( 'img', 1 ) condition allows opening anchor dialog if the anchor is empty (#501).
